x-pack/metricbeat/module/aws/rds/rds.go

```diff
@@ -172,6 +175,27 @@ func getDBInstancesPerRegion(svc rdsiface.ClientAPI) ([]string, map[string]DBDet
 			dbDetails.dbAvailabilityZone = *dbInstance.AvailabilityZone
 		}
 
+		// Get tags for each RDS instance
+		listTagsInput := rds.ListTagsForResourceInput{
+			ResourceName: dbInstance.DBInstanceArn,
+		}
+		reqListTags := svc.ListTagsForResourceRequest(&listTagsInput)
+		outputListTags, err := reqListTags.Send(context.TODO())
+		if err != nil {
+			m.Logger().Warn("ListTagsForResourceRequest failed, rds:ListTagsForResource permission is required for getting tags.")
+			dbDetailsMap[*dbInstance.DBInstanceIdentifier] = dbDetails
```

Comment:
Is it safe to dereference `dbInstance` without having previously checked it for the case of nil? Maybe to add a check on the first part of this loop?

Reply:
Hmmm good point, I'm more worried about `dbInstance` doesn't have DBInstanceIdentifier, DBInstanceArn or DBInstanceClass because that's when we might run into the nil pointer error. I will add checks for that. Thanks!